### ISSUE TITLE
InteropTestsRemoteWithCronet test4MBResponsesAreAccepted

### ISSUE THREAD

**Issue by kpayson64** [grpc/grpc#10106]
Seen on v1.2.x branch:
```
/jenkins/workspace/gRPC_pull_requests_macos/workspace_objc_macos_dbg_native/src/objective-c/tests/InteropTests.m:164: error: -[InteropTestsRemoteWithCronet test4MBResponsesAreAccepted] : ((error) == nil) failed: "Error Domain=io.grpc Code=3 "Received message larger than max (4194316 vs. 4194304)" UserInfo={io.grpc.TrailersKey={
}}" - Finished with unexpected error: Error Domain=io.grpc Code=3 "Received message larger than max (4194316 vs. 4194304)" UserInfo={io.grpc.TrailersKey={
Test Case '-[InteropTestsRemoteWithCronet test4MBResponsesAreAccepted]' failed (5.020 seconds).
Test Suite 'InteropTestsRemoteWithCronet' failed at 2017-03-10 14:01:02.011.
	 Executed 10 tests, with 1 failure (0 unexpected) in 9.259 (27.061) seconds
Test Suite 'InteropTestsRemoteWithCronet.xctest' failed at 2017-03-10 14:01:02.011.
	 Executed 10 tests, with 1 failure (0 unexpected) in 9.259 (27.062) seconds
```
https://grpc-testing.appspot.com/job/gRPC_pull_requests_macos/3368/testReport/junit/(root)/objc_macos_dbg_native/objc_tests/

**Comment by murgatroid99** [comment]
This seems to be the same as #10003 and #9891.

**Comment by muxi** [comment]
Should be fixed by #10042

**Comment by yang-g** [comment]
Dup of #9891
